modin/core/execution/dispatching/factories/factories.py

```diff
@@ -503,17 +503,8 @@ def _read_sql(cls, **kwargs):
         return cls.io_cls.read_sql(**kwargs)
 
 
-@doc(_doc_factory_class, execution_name="experimental PandasOnRay")
-class ExperimentalPandasOnRayFactory(ExperimentalBaseFactory, PandasOnRayFactory):
-    @classmethod
-    @doc(_doc_factory_prepare_method, io_module_name="``ExperimentalPandasOnRayIO``")
-    def prepare(cls):
-        from modin.experimental.core.execution.ray.implementations.pandas_on_ray.io import (
-            ExperimentalPandasOnRayIO,
-        )
-
-        cls.io_cls = ExperimentalPandasOnRayIO
-
+@doc(_doc_abstract_factory_class, role="experimental custom")
+class ExperimentalCustomFactory(BaseFactory):
```

Comment:
Why don't you use `ExperimentalBaseFactory` class as a base class here?

Reply:
Because we inherit `ExperimentalBaseFactory` for multiple other classes (e.g., `ExperimentalPandasOnDaskFactory`) that do not know anything about the methods like `_read_csv_glob`, `_read_pickle_distributed`, etc.

Reply:
Got it. After [this](https://github.com/modin-project/modin/pull/5496) pull request, it will be possible to move experimental functions to the base class.

Reply:
I don't think so. There are other factories that inherit `ExperimentalBaseFactory`.

Reply:
In the rest, we can define the missing methods and explicitly throw `NotImplementedError` exception. Not a big deal.

Reply:
We can even now move all the methods to `ExperimentalBaseFactory` and throw the exception there for the engines other than Ray and Unidist. Do we really want that?

Reply:
I think we want. Thus, we will explicitly say that Modin considers it necessary that such functionality be implemented. The fact that they are not defined for some engine is a temporary problem. Surely they will be implemented in the future. Considering that it is planned to switch to Unidist.

Reply:
I support keeping the _interface_ unified for `modin.pandas.experimental`. If something is not expected to be working under some conditions, we should explicitly let the user know that.

Reply:
Moved the methods to `ExperimentalBaseFactory`.